tools/actions_local_runner.py

```diff
@@ -70,65 +74,119 @@ def find_changed_files() -> List[str]:
     return [x.strip() for x in all_files if x.strip() != ""]
 
 
-async def run_step(step: Dict[str, Any], job_name: str, files: Optional[List[str]], quiet: bool) -> bool:
+def print_results(job_name: str, passed: bool, streams: List[str]) -> None:
+    header(job_name, passed)
+    for stream in streams:
+        stream = stream.strip()
+        if stream != "":
+            print(stream)
+
+
+async def shell_cmd(
+    cmd: str, env: Optional[Dict[str, Any]] = None
+) -> Tuple[bool, str, str]:
+    proc = await asyncio.create_subprocess_shell(
+        cmd,
+        shell=True,
+        cwd=REPO_ROOT,
+        env=env,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        executable=shutil.which("bash"),
+    )
+    stdout, stderr = await proc.communicate()
+
+    return proc.returncode == 0, stdout.decode(), stderr.decode()
+
+
+PASS = color(col.GREEN, "\N{check mark}")
+FAIL = color(col.RED, "x")
+
+
+def header(name: str, passed: bool) -> None:
+    icon = PASS if passed else FAIL
+    print(f"{icon} {color(col.BLUE, name)}")
+
+
+async def run_flake8(files: Optional[List[str]], quiet: bool) -> bool:
+    cmd = "flake8"
+    if files is not None:
+        files_list = " ".join([f"'{f}'" for f in files])
```

Comment:
might it be simpler to use `shlex.join` here? https://docs.python.org/3/library/shlex.html#shlex.join